### PR TITLE
feat: add missing reports queries methods

### PR DESCRIPTION
**Pull request by RiccardoM**
## Description
This PR adds some missing queries methods for the `x/reports` module

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [x] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by dadamu** [comment]
@RiccardoM I thought the reason why these query methods are not in the ADR is they are unnecessary since no one would use the id to query a report or a reason.

**Comment by RiccardoM** [comment]
> @RiccardoM I thought the reason why these query methods are not in the ADR is they are unnecessary since no one would use the id to query a report or a reason. What the reason you decide to add them?

While developing DJuno I discovered these kind of queries might come in handy for third-party tools

**Review by dadamu — APPROVED** [review]

File: proto/desmos/reports/v1/query.proto [hunk 1]
```diff
@@ -18,11 +18,23 @@ service Query {
     option (google.api.http).get = "/desmos/reports/v1/{subspace_id}/reports";
   }
 
+  // Report allows to query the report having the given id
+  rpc Report(QueryReportRequest) returns (QueryReportResponse) {
+    option (google.api.http).get =
+        "/desmos/reports/v1/{subspace_id}/reports/{report_id}";
```
Comment (dadamu) [inline proto/desmos/reports/v1/query.proto:24]:
```suggestion
        "/desmos/reports/v1/subspaces/{subspace_id}/reports/{report_id}";
```

File: proto/desmos/reports/v1/query.proto [hunk 2]
```diff
@@ -18,11 +18,23 @@ service Query {
...
+
   // Reasons allows to query the supported reporting reasons for a subspace
   rpc Reasons(QueryReasonsRequest) returns (QueryReasonsResponse) {
     option (google.api.http).get = "/desmos/reports/v1/{subspace_id}/reasons";
   }
 
+  // Reason allows to query the reason having the given id
+  rpc Reason(QueryReasonRequest) returns (QueryReasonResponse) {
+    option (google.api.http).get =
+        "/desmos/reports/v1/{subspace_id}/reasons/{reason_id}";
```
Comment (dadamu) [inline proto/desmos/reports/v1/query.proto:35]:
```suggestion
        "/desmos/reports/v1/subspaces/{subspace_id}/reasons/{reason_id}";
```

File: proto/desmos/reports/v1/query.proto [hunk 3]
```diff
@@ -18,11 +18,23 @@ service Query {
     option (google.api.http).get = "/desmos/reports/v1/{subspace_id}/reports";
```
Comment (dadamu) [inline proto/desmos/reports/v1/query.proto:18]:
```suggestion
    option (google.api.http).get = "/desmos/reports/v1/subspaces/{subspace_id}/reports";
```

File: proto/desmos/reports/v1/query.proto [hunk 4]
```diff
@@ -18,11 +18,23 @@ service Query {
...
 
+  // Report allows to query the report having the given id
+  rpc Report(QueryReportRequest) returns (QueryReportResponse) {
+    option (google.api.http).get =
+        "/desmos/reports/v1/{subspace_id}/reports/{report_id}";
+  }
+
   // Reasons allows to query the supported reporting reasons for a subspace
   rpc Reasons(QueryReasonsRequest) returns (QueryReasonsResponse) {
     option (google.api.http).get = "/desmos/reports/v1/{subspace_id}/reasons";
```
Comment (dadamu) [inline proto/desmos/reports/v1/query.proto:29]:
```suggestion
    option (google.api.http).get = "/desmos/reports/v1/subspaces/{subspace_id}/reasons";
```

**Review by dadamu — CHANGES_REQUESTED** [review]
We should follow the restful naming rule.

**Review by dadamu — APPROVED** [review]

**Review by bragaz — APPROVED** [review]

**Pull request merged**